File: Clock In/EditNameViewController.m
//
//  EditNameViewController.m
//  Clock In
//
//  Created by Steve Ewell on 12/24/14.
//  Copyright (c) 2014 Steve Ewell. All rights reserved.
//

#import "EditNameViewController.h"

@interface EditNameViewController()

@property (weak, nonatomic) IBOutlet UITextField *editTextField;
@property (strong, nonatomic) NSString *employeeName;
@property (strong, nonatomic) NSUserDefaults *defaults;
@end

@implementation EditNameViewController

-(void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    self.defaults = [[NSUserDefaults alloc]initWithSuiteName:@"group.ewell.TodayExtensionSharingDefaults"];
}

-(void)viewDidLoad {
    [super viewDidLoad];
    
    self.employeeName = [[NSString alloc]init];
    self.editTextField.text = [self.defaults stringForKey:@"employeeName"];
    [self.editTextField becomeFirstResponder];
}

-(void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
    self.employeeName = self.editTextField.text;
    if (self.employeeName.length > 26 || self.employeeName.length == 0) {
        [self.defaults synchronize];
    } else {
        [self.defaults setValue:self.employeeName forKey:@"employeeName"];
        [self.defaults synchronize];
    }
}

-(BOOL)textFieldShouldReturn:(UITextField *)textField {
    self.employeeName = self.editTextField.text;
    if (self.employeeName.length > 26 || self.employeeName.length == 0) {
        UIAlertView *alert = [[UIAlertView alloc]initWithTitle:@"Name is too long or too short!" message:@"Entered names must be between 1 and 26 characters long." delegate:self cancelButtonTitle:@"Ok" otherButtonTitles:nil];
        [alert show];
    } else {
        [self.defaults setValue:self.employeeName forKey:@"employeeName"];
        [self.defaults synchronize];
        [self performSegueWithIdentifier:@"toSettings" sender: self];
    }
    [textField resignFirstResponder];
    return YES;
}

- (void)alertView:(UIAlertView *)alertView didDismissWithButtonIndex:(NSInteger)buttonIndex {
    if (buttonIndex == 0) {
        [self.defaults setValue:@"Name" forKey:@"employeeName"];
        [self.defaults synchronize];
        [self.editTextField becomeFirstResponder];
    }
}

@end

File: Clock In/EditNameViewController.swift
//
//  EditNameViewController.swift
//  Clock In
//
//  Created by Steve Ewell on 11/27/14.
//  Copyright (c) 2014 Steve Ewell. All rights reserved.
//

import UIKit

class EditNameViewController: UITableViewController {
    
    var employeeName:String = String()
    

    @IBOutlet weak var nameEditableCell: SettingsEditableCell!
    override func viewDidLoad() {
        super.viewDidLoad()
        nameEditableCell.cellTextField.text = NSUserDefaults.standardUserDefaults().stringForKey("employeeName")

    }
    
    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(true)
        employeeName = nameEditableCell.cellTextField.text
        if (employeeName.length() > 26 || employeeName.length() == 0) {
            NSUserDefaults.standardUserDefaults().synchronize()
        } else {
            NSUserDefaults.standardUserDefaults().setValue(employeeName, forKey: "employeeName")
            NSUserDefaults.standardUserDefaults().synchronize()
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func textFieldShouldReturn(textField: UITextField) -> Bool {
        employeeName = nameEditableCell.cellTextField.text
        if (employeeName.length() > 26 || employeeName.length() == 0) {
            let alert = UIAlertController(title: "Name is too long or too short!", message: "Entered names must be between 1 and 26 characters long.", preferredStyle: .Alert)
            alert.addAction(UIAlertAction(title: "Ok", style: .Default, handler: alertHandler))
            self.presentViewController(alert, animated: true, completion: nil)
        } else {
            NSUserDefaults.standardUserDefaults().setValue(employeeName, forKey: "employeeName")
            NSUserDefaults.standardUserDefaults().synchronize()
        }
        textField.resignFirstResponder()
        return true
    }
    
    func alertHandler(act:UIAlertAction!) {
        NSUserDefaults.standardUserDefaults().setValue("Name", forKey: "employeeName")
        NSUserDefaults.standardUserDefaults().synchronize()
    }

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using [segue destinationViewController].
        // Pass the selected object to the new view controller.
    }
    */

}
